2025-12-18-Relocate-Files-To-Contained-Package

teskanoo opened the pull request:
Affects Whole Project

Relocates files to create a more self-contained package

All files in custom_templates have been relocated to custom_templates\zenos_ai and all references to import those templates have been updated to look in that subfolder. This reduces the possibility of overwriting user's existing files with the same name and also namespaces the jinja used by zenos.

The following folders are relocated (verbatim) from the root to packages/zenos_ai/xxx
- assets
- automations
- docs
- scripts

The scripts and automations have been updated to start with the expected dictionary key as required for their presence in a package.
- script:
- automation:

All template sensors have been relocated to packages/zenos_ai/sensors (in keeping with the naming conventions already in use).

teskanoo committed:
Affects Whole Project

Relocates files to create a more self-contained package

All files in custom_templates have been relocated to custom_templates\zenos_ai and all references to import those templates have been updated to look in that subfolder. This reduces the possibility of overwriting user's existing files with the same name and also namespaces the jinja used by zenos.

The following folders are relocated (verbatim) from the root to packages/zenos_ai/xxx
- assets
- automations
- docs
- scripts

The scripts and automations have been updated to start with the expected dictionary key as required for their presence in a package.
- script:
- automation:

All template sensors have been relocated to packages/zenos_ai/sensors (in keeping with the naming conventions already in use).

teskanoo committed:
Case sensitivity issue in file name

FIX: Configuration warnings
     Setup of package 'zen_dojoTools_event_emitter' failed: Invalid package definition 'zen_dojoTools_event_emitter': invalid slug zen_dojoTools_event_emitter (try zen_dojotools_event_emitter). Package will not be initialized

nathan-curtis commented:
Tes, we sure we can subdir inside custom_templates?  I'll read these tonight. I assume v. Current on everything?

teskanoo committed:
FIX: Script Zen AdminTools CabinetAdmin failed to set up

Error:
- Only one type can be specified. Found for dictionary value @ data['fields']['auto_profile']['selector']. Got OrderedDict()
- Only one type can be specified. Found for dictionary value @ data['fields']['cabinet_guid']['selector']. Got OrderedDict()
- Only one type can be specified. Found for dictionary value @ data['fields']['confirm_action']['selector']. Got OrderedDict()
- Only one type can be specified. Found for dictionary value @ data['fields']['flags_override']['selector']. Got OrderedDict()
- Only one type can be specified. Found for dictionary value @ data['fields']['force_new_guid']['selector']. Got OrderedDict()
- Only one type can be specified. Found for dictionary value @ data['fields']['mount_scan']['selector']. Got OrderedDict()
- Only one type can be specified. Found for dictionary value @ data['fields']['repair_acls']['selector']. Got OrderedDict()
- Only one type can be specified. Found for dictionary value @ data['fields']['repair_flags_bits']['selector']. Got OrderedDict() Only one type can be specified. Found for dictionary value @ data['fields']['repair_flags_full']['selector']. Got OrderedDict()
- Only one type can be specified. Found for dictionary value @ data['fields']['repair_flags_profile']['selector']. Got OrderedDict()
- Only one type can be specified. Found for dictionary value @ data['fields']['repair_guid']['selector']. Got OrderedDict()
- Only one type can be specified. Found for dictionary value @ data['fields']['repair_labels']['selector']. Got OrderedDict()
- Only one type can be specified. Found for dictionary value @ data['fields']['repair_mounts']['selector']. Got OrderedDict()
- Only one type can be specified. Found for dictionary value @ data['fields']['repair_volumeinfo']['selector']. Got OrderedDict().

teskanoo committed:
FIX: Script Zen DojoTools Index (RC1) failed to set up (and many others)

Most of these issues stem from the way the selectors are defined in the main branch YAML - Sadly HA supports thinks like the following in the UI but not in YAML

```YAML
        selector:
          boolean: {}
```

The other issues were just to do with case sensitivity issues in the filenames of the YAML files (they must be fully lowercase) for YAML packages.

Fixes the following and about 30 more...

Error:
- Only one type can be specified. Found for dictionary value @ data['fields']['expand_entities']['selector']. Got OrderedDict()
- Only one type can be specified. Found for dictionary value @ data['fields']['index_command']['selector']. Got OrderedDict()
- Only one type can be specified. Found for dictionary value @ data['fields']['label_1']['selector']. Got OrderedDict()
- Only one type can be specified. Found for dictionary value @ data['fields']['label_2']['selector']. Got OrderedDict()
- Only one type can be specified. Found for dictionary value @ data['fields']['strict']['selector']. Got OrderedDict()

nathan-curtis commented:
Ah cool that's my default Ai task. I THINK I can default that out considering I recommended Spook I need to be considerate.  

can you bug it. I'm inthe middle of finishing something so I can get back to the installs 

Also are you in a position to try and format that cabinet set if I give you a script?

My next task is tk stand up a vanilla brand new HA and from that drop our package and following 'easy' instructions get someone from (paste) to fully running construct.

teskanoo commented:
> Also are you in a position to try and format that cabinet set if I give you a script?

I'll give it a go, if you can provide more clarity on what you mean by "format that cabinet set"

teskanoo committed:
FIX: The script "Zen DojoTools FileCabinet (RC1)" (script.zen_dojotools_filecabinet) is not active because the configuration has errors.

     Error:Only one type can be specified. Found for dictionary value @ data['fields']['destination_cabinet']['selector']. Got OrderedDict().

nathan-curtis commented:
Let me finish some of the edits I have to finish. Then we will merge it over into the new path.

All the scripts and automations will take an edit to the file when we do. That makes them incomparable with what my current edits are. As soon S I finish I'll move them but I can't until.
